.github/workflows/ci.yml

```diff
@@ -17,8 +17,8 @@ on:
 
 env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  JAVA_OPTS: '-Xms15G -Xmx15G -Xss10M -XX:MaxMetaspaceSize=2G -XX:ReservedCodeCacheSize=256M -Dfile.encoding=UTF-8'
-  JVM_OPTS: '-Xms15G -Xmx15G -Xss10M -XX:MaxMetaspaceSize=2G -XX:ReservedCodeCacheSize=256M -Dfile.encoding=UTF-8'
+  JDK_JAVA_OPTIONS: '-Xms8G -Xmx8G -XX:+UseG1GC -Xss10M -XX:MaxMetaspaceSize=2G -XX:ReservedCodeCacheSize=256M -Dfile.encoding=UTF-8'
```

Comment:
I think 15G was way too high so I reduced it to 8G. I think this is OK given that CI was using 2G till now